ArticleTemplates/assets/js/modules/cards.js

```diff
@@ -1,96 +1,52 @@
 define([
-    'flipSnap',
     'modules/util'
-],
+], 
 function (
-    flipSnap,
     util
 ) {
     'use strict';
 
     var initialised = false;
+    var existingRelatedContentPosition;
 
-    function setupGlobals() {
-        window.articleCardsInserter = articleCardsInserter;
-        window.articleCardsFailed = articleCardsFailed;
-
-        window.applyNativeFunctionCall('articleCardsInserter');
-        window.applyNativeFunctionCall('articleCardsFailed');
-    }
-
-    function articleCardsInserter(html) {
-        var relatedContentWrapper = document.getElementsByClassName('related-content__wrapper')[0];
-
-        if (relatedContentWrapper) {
-            if (!html) {
-                articleCardsFailed();
-            } else {
-                relatedContentWrapper.innerHTML = html;
-                snapToGrid();
-            }
+    function ready(config) {
+        if (!initialised && GU.opts.platform === 'ios') {
+            initialised = true;
+            new window.MutationObserver(function(mutations) {
+                var newRelatedContentPosition = getRelatedContentPosition();
+                if (newRelatedContentPosition !== existingRelatedContentPosition) {
+                    existingRelatedContentPosition = newRelatedContentPosition;
```

Comment:
OK this comparison will always return false as you can't test for (deep) equality between to objects like this. You have 2 options here. Iterate over each property of the 2 objects to see if there are any differences. Or stringifying both objects will work in this case: `JSON.stringify(newRelatedContentPosition) !== JSON.stringify(existingRelatedContentPosition)`